routing/datasource.go

```diff
@@ -519,6 +536,11 @@ func processRouteDefs(o *Options, defs []*eskip.Route) (routes []*Route, invalid
 		} else {
 			invalidDefs = append(invalidDefs, def)
 			o.Log.Errorf("failed to process route %s: %v", def.Id, err)
+
+			var defErr invalidDefinitionError
+			if errors.As(err, &defErr) && o.Metrics != nil {
+				o.Metrics.IncInvalidRoutes(defErr.Code())
+			}
```

Comment:
Here I think we should have else branch to count other types of errors (that may appear in the future) with an "other" or "generic" error code.